DatadogSessionReplay/SRExample/SRSnapshotTests/Utils/ImageComparison.swift

```diff
@@ -0,0 +1,249 @@
+///
+/// - Parameters:
+///   - image: the image to compare OR record
+///   - referenceImage: the reference file to compare against
+///   - record: if `true`, then reference file will be created / overwritten with `image` data
+///   - file: `#filePath` for eventual `XCTest` assertion failure
+///   - line: `#line` for eventual `XCTest` assertion failure
+internal func compare(
+    image: UIImage,
+    referenceImage: ReferenceImage,
+    record: Bool,
+    file: StaticString = #filePath,
+    line: UInt = #line
+) throws {
+    let simulatorModel = ProcessInfo.processInfo.environment["SIMULATOR_MODEL_IDENTIFIER"]
+    let osVersion = UIDevice.current.systemVersion
+
+    guard simulatorModel == "iPhone14,7", osVersion == "16.2" else {
+        XCTFail(
+            "Snapshots must be compared on iPhone 14 Simulator (iPhone14,7) + iOS 16.2. " +
+            "Running on \(simulatorModel ?? "unknown") + iOS \(osVersion) instead.",
+            file: file,
+            line: line
+        )
+        return
+    }
+
+    if record {
+        let directoryURL = referenceImage.url.deletingLastPathComponent()
+        try FileManager.default.createDirectory(at: directoryURL, withIntermediateDirectories: true)
+        let data = try XCTUnwrap(image.pngData(), "Failed to create `pngData()` for `image`", file: file, line: line)
+        try data.write(to: referenceImage.url)
+    } else {
+        let oldImageData = try Data(contentsOf: referenceImage.url)
+        let oldImage = try XCTUnwrap(UIImage(data: oldImageData), "Failed to read reference image", file: file, line: line)
+
+        // Check if both images are identical (precission: 1) or their difference is not
+        // noticable for the human eye (perceptualPrecision: 0.98).
+        // Ref.: http://zschuessler.github.io/DeltaE/learn/#toc-defining-delta-e
+        if let difference = compare(oldImage, image, precision: 1, perceptualPrecision: 0.98) {
+            XCTFail("\(difference) (in file: \(referenceImage.url)", file: file, line: line)
+        }
+    }
+}
+
+// Copyright © pointfreeco swift-snapshot-testing (MIT License)
+// See license: https://github.com/pointfreeco/swift-snapshot-testing/blob/main/LICENSE
+//
+// This code was taken from https://github.com/pointfreeco/swift-snapshot-testing and
+// modified for the purpose of testing Framer lib (© ncreated 2022).
+//
+// Modifications made:
+// - simplification of platform & version specific code branches
+import UIKit
+import CoreImage.CIKernel
+import MetalPerformanceShaders
+
+// MARK: - From `swift-snapshot-testing` /Sources/SnapshotTesting/Snapshotting/UIImage.swift
+
+// remap snapshot & reference to same colorspace
+private let imageContextColorSpace = CGColorSpace(name: CGColorSpace.sRGB)
+private let imageContextBitsPerComponent = 8
+private let imageContextBytesPerPixel = 4
+
+func compare(_ old: UIImage, _ new: UIImage, precision: Float, perceptualPrecision: Float) -> String? {
```

Comment:
With `_DDEvaluateAssertion ` public, this could be turned into a `DDAssertImageComparison`.